ci: Clone entire bitcoin-core/qa-assets repo only when run fuzzing

hebasto opened the pull request:
This PR speeds up CI tasks that run unit tests but do not run fuzzing.

On my machine:
```
$ time git clone --depth=1 https://github.com/bitcoin-core/qa-assets
Cloning into 'qa-assets'...
remote: Enumerating objects: 289750, done.
remote: Counting objects: 100% (289750/289750), done.
remote: Compressing objects: 100% (207687/207687), done.
remote: Total 289750 (delta 16863), reused 275449 (delta 12092), pack-reused 0
Receiving objects: 100% (289750/289750), 1.39 GiB | 4.79 MiB/s, done.
Resolving deltas: 100% (16863/16863), done.
Updating files: 100% (294515/294515), done.

real	7m43,417s
user	2m39,771s
sys	0m43,272s
```

hebasto committed:
ci: Clone entire bitcoin-core/qa-assets repo only when run fuzzing

MarcoFalke commented:
Once we drop bionic, this can be replaced by:

```
git clone --depth 1 --filter=blob:none --sparse https://github.com/bitcoin-core/qa-assets
cd qa-assets/
git sparse-checkout set unit_test_data

MarcoFalke commented on ci/test/04_install.sh:
```diff
@@ -87,13 +87,20 @@ fi
   fi
 
   export DIR_FUZZ_IN=${DIR_QA_ASSETS}/fuzz_seed_corpus/
   export DIR_UNIT_TEST_DATA=${DIR_QA_ASSETS}/unit_test_data/
```
Is this still needed?

hebasto replied:
Yes, in case both `"$RUN_FUZZ_TESTS" = "true"` and `"$RUN_UNIT_TESTS" = "true"`.

MarcoFalke replied:
It is not possible to run both.

Fuzz tests can only run with libfuzzer (right now), in which case unit tests don't work and vice-versa

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/23938#issuecomment-1006633606).

hebasto commented:
Updated 2db1fe3dcd9fc5b2a95f6e5e6ed63d8625b16d6d -> 0b7c55f157514e1845934fff3f9c66e862c526ca ([pr23938.01](https://github.com/hebasto/bitcoin/commits/pr23938.01) -> [pr23938.02](https://github.com/hebasto/bitcoin/commits/pr23938.02), [diff](https://github.com/hebasto/bitcoin/compare/pr23938.01..pr23938.02)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/23938#discussion_r779579118)

MarcoFalke commented:
cr ACK 0b7c55f157514e1845934fff3f9c66e862c526ca
